src/main/java/fr/openent/lystore/service/impl/DefaultEquipmentService.java

```diff
@@ -96,19 +100,57 @@ public void listEquipments(Integer page, String order, Boolean reverse, List<Str
+        String query = "SELECT equip.reference, equip.name, equip.id, equip.status,  equip.price, supplier.name as supplier_name, " +
+                "contract.name as contract_name, tax.value as tax_amount ,  array_to_json(array_agg(opts.*)) as options " +
                 "FROM " + Lystore.lystoreSchema + ".equipment equip " +
                 "INNER JOIN " + Lystore.lystoreSchema + ".tax ON tax.id = equip.id_tax " +
                 "INNER JOIN " + Lystore.lystoreSchema + ".contract ON (contract.id = equip.id_contract) " +
                 "INNER JOIN " + Lystore.lystoreSchema + ".supplier ON (contract.id_supplier = supplier.id) " +
+                "LEFT JOIN " + Lystore.lystoreSchema + ".equipment_option ON (equip.id = equipment_option.id_equipment) " +
+                "LEFT JOIN ( " +
+                "SELECT equipment.id as id_equipment, equipment.reference, equipment.id_type, equipment_option.id," +
+                " equipment_option.id_option, equipment.name, equipment.price, equipment_option.amount, " +
+                "equipment_option.required, tax.value as tax_amount, equipment_option.id_equipment as master_equipment," +
+                " equipment_type.name as nametype " +
+                "FROM " + Lystore.lystoreSchema + ".equipment " +
+                "INNER JOIN " + Lystore.lystoreSchema + ".tax  ON (equipment.id_tax = tax.id) " +
+                "INNER JOIN " + Lystore.lystoreSchema + ".equipment_option  ON (equipment_option.id_option = equipment.id) " +
+                "INNER JOIN " + Lystore.lystoreSchema + ".equipment_type ON (equipment_type.id = equipment.id_type) " +
+                ") opts ON (equipment_option.id_option = opts.id_equipment AND opts.master_equipment = equip.id) " +
                 getTextFilter(filters, request) +
+                " GROUP BY equip.reference ,    equip.name," +
+                "       equip.id," +
+                "       equip.status," +
+                "       equip.price, " +
+                "       supplier_name, " +
+                "       contract_name,  " +
+                "       tax.value   " +
                 "ORDER by " + getSqlOrderValue(order) + " " + getSqlReverseString(reverse);
 
         if (page != null) {
             query += " LIMIT " + Lystore.PAGE_SIZE + " OFFSET ?";
             params.add(Lystore.PAGE_SIZE * page);
         }
-        sql.prepared(query, params, SqlResult.validResultHandler(handler));
+        sql.prepared(query, params, SqlResult.validResultHandler(event -> {
+            if (event.isRight()) {
+                handler.handle(new Either.Right<>(new JsonArray(event.right().getValue().stream()
+                        .filter(JsonObject.class::isInstance)
+                        .map(JsonObject.class::cast)
+                        .map(elem -> {
```

Comment:
en vrai:
```suggestion
                        .map(elem -> {
                            JsonObject.cast(elem).put(LystoreBDD.OPTIONS, new JsonArray(elem.getString(LystoreBDD.OPTIONS)));
                                return  elem;
                         })
```
ça enlève une map